great_expectations/data_context/store/checkpoint_store.py

```diff
@@ -224,3 +228,65 @@ def create(self, checkpoint_config: CheckpointConfig) -> Optional[DataContextKey
         config = self.get(key=key)
 
         return config
+
+
+# NOTE: Will eventually be promoted to 'CheckpointStore' once the legacy Checkpoint
+#       and its related classes are removed
+class V1CheckpointStore(Store):
+    _key_class = StringKey
```

Comment:
Follows the exact same model as the `ValidationConfigStore` I wrote recently.

Once this is in, I will take the following steps:
- Delete the legacy checkpoint, checkpoint config, checkpoint store, and store validation result action
- Wire up this new store with the checkpoint factory